kubexecutor/src/main/kotlin/pl/edu/prz/kod/executor/AbstractExecutor.kt

```diff
@@ -2,9 +2,27 @@ package pl.edu.prz.kod.executor
 
 import pl.edu.prz.kod.domain.ExecutionResult
 import pl.edu.prz.kod.domain.Language
+import pl.edu.prz.kod.error.ProcessTimedOutError
+import java.io.File
 import java.util.concurrent.TimeUnit
 
 sealed class AbstractExecutor(language: Language) {
-    abstract fun execute(code: String, timeout: Long, timeoutUnit: TimeUnit): ExecutionResult
+    private val timeout = 5L
+    private val timeoutUnit = TimeUnit.SECONDS
+
+    abstract fun execute(code: String): ExecutionResult
+
+    fun runSystemCommand(command: String, workDir: File? = null): Process {
```

Comment:
I'm not really convinced with this nullable `workDir`, how about adding it as mandatory with some default value, i.e. `/app/executor`?

Reply:
I've intentionally put default value of `null` there, because if `null` is passed to `directory` method (a few lines below) then the new process will use working directory of current process (`/app` in this case) and that's exactly what I wanted to achieve. I changed the default value of this argument to `/app` so it still has the same effect but argument is non-nullable.